pkg/commands/workload_apply.go

```diff
@@ -158,20 +152,22 @@ func (opts *WorkloadApplyOptions) Exec(ctx context.Context, c *cli.Config) error
 		return nil
 	}
 
+	workloadExists := currentWorkload != nil
+
+	if okToPush, err := opts.PublishLocalSource(ctx, c, currentWorkload, workload, shouldPrint); err != nil {
+		return err
+	} else if !okToPush {
+		return nil
+	}
+	opts.ManageLocalSourceProxyAnnotation(currentWorkload, workload)
```

Comment:
*Question:* Could this be moved to `workload` (common) instead of calling this in 3 different functions?

Reply:
it didn't seem as the best decision for me, because the other part where we could call this func is inside `PublishLocalSource()` and, for me, this bit of the annotation is not a concern for that func, so I decided to add it in exec despite it requires a bit of repetition.